src/components/DiskSource/DiskSource.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { V1alpha1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  Checkbox,
+  FormGroup,
+  Popover,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextInput,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import { PersistentVolumeClaimSelect } from '../PersistentVolumeClaimSelect';
+
+import {
+  DISK_SOURCE_OPTIONS,
+  HTTP_DISK_SOURCE_NAME,
+  PVC_DISK_SOURCE_NAME,
+  PVC_SIZE_FORMATS,
+  REGISTRY_DISK_SOURCE_NAME,
+} from './constants';
+import { getPVCDiskSource } from './utils';
+import { VolumeSize } from './VolumeSize';
+import { DISK_SOURCE_OPTIONS_IDS } from '.';
+
+import './DiskSource.scss';
+
+type DiskSourceProps = {
+  template: V1Template;
+  onChange: (diskSource: any) => void;
+};
+
+export const DiskSource: React.FC<DiskSourceProps> = ({ onChange }) => {
```

Comment:
you should work only withing the extension you are currently working on, so if you need a component it's in:
`/<the extentions you work on>/components/<your component>`
if you think it should be shared you can move it to the components repo

Reply:
Do you think moving DiskSource and PersistenceVolumeNameAndNamespace (PersistentVolumeClainSelect in this pr)   components in the components repo could be a good idea?